packages/keyring-controller/src/KeyringController.ts

```diff
@@ -664,7 +667,7 @@ export class KeyringController extends BaseController<
     password: string,
     seed: Uint8Array,
   ): Promise<KeyringControllerMemState> {
-    const releaseLock = await this.mutex.acquire();
+    const releaseLock = await this.#initVaultMutex.acquire();
```

Comment:
Hmm. I've just noticed that if the user has an invalid password, this lock would never be released.

This is a pre-existing bug though, we can address it separately

Reply:
This will get fixed in this PR https://github.com/MetaMask/core/pull/4156, because throwing the error will release it